### ISSUE TITLE
Ability for Consul to serve traffic despite an optional Health Check failure 

### ISSUE THREAD

**Issue by teeto11** [hashicorp/nomad#23504]
**Proposal**
Hello it will be nice to have a way to configure health checks for certain services as **optional** , so when this service fails, the app still works on the failure of just that particular service. Does something like this exist and i may have missed it ?

**Use-cases**
For example if a health check for a service is marked as not healthy on consul, i want to be able to still serve traffic to the app because that service isn't a major dependency to cause a restart.

**Comment by teeto11** [comment]
Just like this feature [issue](https://github.com/hashicorp/nomad/issues/876) but this should not stop consul from still serving the app

**Comment by pkazmierczak** [comment]
Hi @teeto11, you can set [on_update](https://developer.hashicorp.com/nomad/docs/job-specification/check#on_update) field to `ignore_warnings` or `ignore` to instruct Nomad to treat the service as healthy regardless of its Consul status. Is that what you meant? let me know if this works for you.

**Comment by pkazmierczak** [comment]
right, so `ignore_warnings` is a Nomad setting, it will only affect how Nomad will proceed with service deployment. If you don't want the check to register on Consul as `critical`, you can use a `script` check type or perhaps set high values for `failures_before_critical` or `timeout` fields? Or disable the check altogether?

**Comment by teeto11** [comment]
Thanks @pkazmierczak , where can i find the check type `script`

**Comment by pkazmierczak** [comment]
We have some examples in [our documentation](https://developer.hashicorp.com/nomad/docs/job-specification/check#script-checks-with-shells) and there's more examples on [Consul website](https://developer.hashicorp.com/consul/docs/services/usage/checks#script-checks)

**Comment by pkazmierczak** [comment]
I'll close this issue for now, feel free to re-open if you're still having trouble.
